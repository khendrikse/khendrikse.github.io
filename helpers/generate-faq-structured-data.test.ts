import generateFaqStructuredData from './generate-faq-structured-data';

describe('generateFaqStructuredData', () => {
  it('generates faq structured data', () => {
    const input = [{ question: 'a question', answer: 'an answer' }];
    const result = generateFaqStructuredData(input);
    expect(result).toMatchSnapshot();
  });

  it('returns null without input', () => {
    expect(generateFaqStructuredData()).toBe(null);
  });
});
