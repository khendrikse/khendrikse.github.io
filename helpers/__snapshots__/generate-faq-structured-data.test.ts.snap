// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateFaqStructuredData generates faq structured data 1`] = `"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"FAQPage\\",\\"mainEntity\\":[[{\\"@type\\":\\"Question\\",\\"name\\":\\"a question\\",\\"acceptedAnswer\\":{\\"@type\\":\\"Answer\\",\\"text\\":\\"an answer\\"}}]]}"`;
